docker_cli/_exceptions.py

```diff
@@ -0,0 +1,125 @@
+from typing import Optional, overload
+
+
+class CommandNotFoundError(Exception):
+    """Raised when a command is attempted but not found."""
+
+    def __init__(self, command_name: str):
+        """
+        Raise this exception when a command is attempted but not found.
+
+        Parameters
+        ----------
+        command_name : str
+            The name of the command.
+        """
+        self.command_name = command_name
+        super(CommandNotFoundError, self).__init__(
+            f"Command '{self.command_name}' was run, but is not "
+            "present on the Docker image."
+        )
+
+
+class DockerBuildError(Exception):
+    """Raised when a docker image fails to build"""
+
+    @overload
+    def __init__(self, *, message: Optional[str]):
```

Comment:
`message` isn't optional in this overload. The fact that we're using `None` as the default value is just an implementation detail of how we implement method overloading.

```suggestion
    def __init__(self, *, message: str):
```